src/rules/noMagicNumbersRule.ts

```diff
@@ -63,44 +61,31 @@ export class Rule extends Lint.Rules.AbstractRule {
     public static DEFAULT_ALLOWED = [ -1, 0, 1 ];
 
     public apply(sourceFile: ts.SourceFile): Lint.RuleFailure[] {
-        return this.applyWithWalker(new NoMagicNumbersWalker(sourceFile, this.getOptions()));
+        const options = this.getOptions();
+        const ruleArguments = options.ruleArguments;
+        const allowedNumbers = ruleArguments.length > 0 ? ruleArguments : Rule.DEFAULT_ALLOWED;
+        return this.applyWithWalker(new NoMagicNumbersWalker(sourceFile, options.ruleName, new Set(allowedNumbers.map(String))));
```

Comment:
I like this concept of parsing rule options as part of the rule instead of part of the rule implementation - seems like a better separation of concerns to me.